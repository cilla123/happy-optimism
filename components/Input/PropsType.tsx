export interface BaseInputProps {
  type?: string;
  placeholder?: string;
  disabled?: boolean;
  maxLength?: number;
  focused?: boolean;
  autoFocus?: boolean;
  style?: React.CSSProperties;
}

export interface BaseInputTextProps extends BaseInputProps {
  value?: string;
  defaultValue?: string;
  onChange?: (value?: string) => void;
  onFocus?: (value?: string) => void;
  onBlur?: (value?: string) => void;
}

export interface BaseInputNumberProps extends BaseInputProps {
  type?: 'number' | 'price' | 'idcard';
  value?: string | number;
  defaultValue?: string | number;
  onChange?: (value?: string | number) => void;
  onFocus?: (value?: string | number) => void;
  onBlur?: (value?: string | number) => void;
}

export interface BaseInputTextareaProps extends BaseInputProps {
  value?: string;
  defaultValue?: string;
  rows?: number;
  autoHeight?: boolean;
  showLength?: boolean;
  onChange?: (value?: string) => void;
  onFocus?: (value?: string) => void;
  onBlur?: (value?: string) => void;
}
