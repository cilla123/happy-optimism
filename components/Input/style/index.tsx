import '../../style/';
import '../../KeyboardPicker/style/';
import './index.scss';
