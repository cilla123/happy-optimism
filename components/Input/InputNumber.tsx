import React, { Component } from 'react';
import { findDOMNode } from 'react-dom';
import classnames from 'classnames';
import { BaseInputNumberProps } from './PropsType';
import Events from '../utils/events';
import KeyboardPicker from '../KeyboardPicker';

declare const document;

export interface InputNumberProps extends BaseInputNumberProps {
  prefixCls?: string;
  className?: string;
}

export default class InputNumber extends Component<InputNumberProps, any> {

  static defaultProps = {
    prefixCls: 'za-input',
    disabled: false,
  };

  private content;
  private picker;

  constructor(props) {
    super(props);
    this.state = {
      value: props.value || props.defaultValue || '',
      visible: props.focused || false,
    };
  }

  componentDidMount() {
    Events.on(document.body, 'click', this.onMaskClick);
    if (this.props.autoFocus || this.state.focused) {
      this.onFocus();
    }
  }

  componentWillReceiveProps(nextProps) {
    if ('focused' in nextProps || 'autoFocus' in nextProps) {
      if (nextProps.focused || nextProps.autoFocus) {
        this.onFocus();
      } else {
        this.onBlur();
      }
    }
  }

  componentWillUnmount() {
    Events.off(document.body, 'click', this.onMaskClick);
  }

  onMaskClick = (e) => {
    if (!this.state.visible) {
      return;
    }

    const cNode = ((node) => {
      while (node.parentNode && node.parentNode !== document.body) {
        if (node === findDOMNode(this.picker)) {
          return node;
        }
        node = node.parentNode;
      }
    })(e.target);

    if (!cNode) {
      this.onBlur();
    }
  }

  onKeyClick = (key) => {
    if (['close', 'ok'].indexOf(key) > -1) {
      this.onBlur();
      return;
    }
    const value = this.state.value;
    const newValue = (key === 'delete')
      ? value.slice(0, value.length - 1)
      : value + key;

    if (newValue !== value) {
      const { onChange } = this.props;
      this.setState({ value: newValue }, () => this.scrollToEnd());

      if (typeof onChange === 'function') {
        onChange(newValue);
      }
    }
  }

  scrollToStart = () => {
    this.content.scrollLeft = 0;
  }

  scrollToEnd = () => {
    this.content.scrollLeft = this.content.scrollWidth;
  }

  onFocus = () => {
    if (this.state.visible) {
      return;
    }

    this.setState({ visible: true });
    this.scrollToEnd();
    const { onFocus } = this.props;
    if (typeof onFocus === 'function') {
      onFocus(this.state.value);
    }
  }

  onBlur = () => {
    if (!this.state.visible) {
      return;
    }

    this.setState({ visible: false });
    this.scrollToStart();
    const { onBlur } = this.props;
    if (typeof onBlur === 'function') {
      onBlur(this.state.value);
    }
  }

  render() {
    const { prefixCls, className, type, disabled, placeholder } = this.props;
    const { visible, value } = this.state;

    const cls = classnames(prefixCls, `${prefixCls}-number`, className, {
      disabled,
      focus: visible,
    });

    return (
      <div className={cls} onClick={this.onFocus}>
        {!value && <div className={`${prefixCls}-placeholder`}>{placeholder}</div>}
        <div className={`${prefixCls}-content`} ref={(ele) => { this.content = ele; }}>{value}</div>
        <input
          type="hidden"
          value={value}
          disabled={disabled}
        />
        <KeyboardPicker
          ref={(ele) => { this.picker = ele; }}
          visible={visible}
          type={type}
          onKeyClick={this.onKeyClick}
        />
      </div>
    );
  }
}
