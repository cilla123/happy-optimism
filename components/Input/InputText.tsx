import React, { PureComponent } from 'react';
import classnames from 'classnames';
import { BaseInputTextProps } from './PropsType';

export interface InputTextProps extends BaseInputTextProps {
  prefixCls?: string;
  className?: string;
}

export default class InputText extends PureComponent<InputTextProps, any> {

  static defaultProps = {
    prefixCls: 'za-input',
    disabled: false,
    focused: false,
  };

  private input;

  constructor(props) {
    super(props);
    this.state = {
      focused: props.focused || false,
    };
  }

  componentDidMount() {
    if (this.props.autoFocus || this.state.focused) {
      this.input.focus();
    }
  }

  componentWillReceiveProps(nextProps) {
    if ('focused' in nextProps) {
      this.setState({
        focused: nextProps.focused,
      });
    }
  }

  componentDidUpdate() {
    if (this.state.focused) {
      this.input.focus();
    }
  }

  onFocus = (e) => {
    if (!('focused' in this.props)) {
      this.setState({
        focused: true,
      });
    }

    const { onFocus } = this.props;
    if (typeof onFocus === 'function') {
      onFocus(e.target.value);
    }
  }

  onBlur = (e) => {
    if (!('focused' in this.props)) {
      this.setState({
        focused: false,
      });
    }
    const { onBlur } = this.props;
    if (typeof onBlur === 'function') {
      onBlur(e.target.value);
    }
  }

  onChange = (e) => {
    const { onChange } = this.props;
    if (typeof onChange === 'function') {
      onChange(e.target.value);
    }
  }

  render() {
    const { prefixCls, className, disabled, focused, ...others } = this.props;
    const cls = classnames(prefixCls, `${prefixCls}-text`, className, {
      disabled,
    });

    return (
      <div className={cls}>
        <input
          {...others}
          ref={(ele) => { this.input = ele; }}
          type="text"
          disabled={disabled}
          onChange={this.onChange}
          onFocus={this.onFocus}
          onBlur={this.onBlur}
        />
      </div>
    );
  }
}
