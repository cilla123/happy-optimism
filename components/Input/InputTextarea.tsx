import React, { PureComponent } from 'react';
import classnames from 'classnames';
import { BaseInputTextareaProps } from './PropsType';

const regexAstralSymbols = /[\uD800-\uDBFF][\uDC00-\uDFFF]|\n/g;

const countSymbols = (text = '') => {
  return text.replace(regexAstralSymbols, '_').length;
};

export interface InputTextareaProps extends BaseInputTextareaProps {
  prefixCls?: string;
  className?: string;
}

export default class InputTextarea extends PureComponent<InputTextareaProps, any> {

  static defaultProps = {
    prefixCls: 'za-input',
    disabled: false,
    autoHeight: false,
    showLength: false,
    focused: false,
  };

  private input;

  constructor(props) {
    super(props);
    this.state = {
      length: (props.value || props.defaultValue || '').length,
      focused: props.focused || false,
    };
  }

  componentDidMount() {
    if (this.props.autoFocus || this.state.focused) {
      this.input.focus();
    }
  }

  componentDidUpdate() {
    const { autoHeight } = this.props;
    if (autoHeight) {
      this.input.style.height = '';
      this.input.style.height = `${this.input.scrollHeight}px`;
    }
    if (this.state.focused) {
      this.input.focus();
    }
  }

  componentWillReceiveProps(nextProps) {
    if ('focused' in nextProps) {
      this.setState({
        focused: nextProps.focused,
      });
    }
  }

  onInputFocus = (e) => {
    if (!('focused' in this.props)) {
      this.setState({
        focused: true,
      });
    }

    const { onFocus } = this.props;
    if (typeof onFocus === 'function') {
      onFocus(e.target.value);
    }
  }

  onInputBlur = (e) => {
    if (!('focused' in this.props)) {
      this.setState({
        focused: false,
      });
    }
    const { onBlur } = this.props;
    if (typeof onBlur === 'function') {
      onBlur(e.target.value);
    }
  }

  onInputChange = (e) => {
    const { onChange } = this.props;
    const value = e.target.value;
    const length = countSymbols(value) + (value ? value.length : 0);
    this.setState({ length });
    if (typeof onChange === 'function') {
      onChange(value);
    }
  }

  render() {
    const {
      prefixCls,
      className,
      maxLength,
      disabled,
      autoHeight,
      showLength,
      focused,
      ...others,
    } = this.props;

    const cls = classnames(prefixCls, `${prefixCls}-textarea`, className, {
      disabled,
    });

    const textLengthRender =
      showLength &&
      maxLength &&
      <div className={`${prefixCls}-length`}>
        {`${this.state.length}/${maxLength}`}
      </div>;

    return (
      <div className={cls}>
        <textarea
          {...others}
          ref={(ele) => { this.input = ele; }}
          disabled={disabled}
          maxLength={maxLength}
          onChange={this.onInputChange}
          onFocus={this.onInputFocus}
          onBlur={this.onInputBlur}
        />
        {textLengthRender}
      </div>
    );
  }
}
