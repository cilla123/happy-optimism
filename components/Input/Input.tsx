import React, { PureComponent } from 'react';
import { BaseInputProps } from './PropsType';
import InputNumber from './InputNumber';
import InputText from './InputText';
import InputTextarea from './InputTextarea';

export interface InputProps extends BaseInputProps {}

export default class Input extends PureComponent<InputProps, {}> {

  static defaultProps = {
    type: 'text',
  };

  render() {
    const { type, ...others } = this.props;

    switch (type) {
      case 'idcard':
      case 'price':
      case 'number':
        return <InputNumber {...others} type={type} />;

      case 'textarea':
        return <InputTextarea {...others} />;

      default:
        return <InputText {...others} type={type} />;
    }
  }
}
