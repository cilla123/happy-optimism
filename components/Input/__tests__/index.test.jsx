import React from 'react';
import { render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Input from '../index';

describe('Input', () => {
  it('renders correctly', () => {
    const wrapper = render(<Input type="text" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('showLength', () => {
    const wrapper = render(<Input showLength maxLength={100} type="textarea" rows={4} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('Input.Textarea', () => {
  it('renders correctly', () => {
    const wrapper = render(<Input type="textarea" rows={4} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('autoHeight', () => {
    jest.useFakeTimers();
    const props = {
      autoHeight: true,
      type: 'textarea',
      rows: 4,
      value: 'foo',
      onChange: jest.fn(),
    };
    const wrapper = mount(<Input {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.find('textarea').simulate('change', { target: { value: 'this is a test!' } });
    expect(props.onChange).toBeCalledWith('this is a test!');
    jest.runAllTimers();
    wrapper.unmount();
  });
});

describe('Input.Number', () => {
  it('renders correctly', () => {
    const wrapper = mount(<Input type="number" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

//   it('enter number', () => {
//     const onChange = jest.fn();
//     const wrapper = mount(<Input type="number" onChange={onChange} />);
//     wrapper.find('input').simulate('focus');
//     const keys = wrapper.find('.za-keyboard-keys');
//     keys.childAt(0).simulate('click');
//     expect(onChange).toBeCalledWith('1');
//     wrapper.unmount();
//   });
});

describe('Input.Price', () => {
  it('renders correctly', () => {
    const wrapper = mount(<Input type="price" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('Input.Idcard', () => {
  it('renders correctly', () => {
    const wrapper = mount(<Input type="idcard" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
