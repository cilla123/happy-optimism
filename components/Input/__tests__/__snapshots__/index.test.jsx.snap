// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<div
  class="za-input za-input-text"
>
  <input
    type="text"
  />
</div>
`;

exports[`Input showLength 1`] = `
<div
  class="za-input za-input-textarea"
>
  <textarea
    maxlength="100"
    rows="4"
  />
  <div
    class="za-input-length"
  >
    0/100
  </div>
</div>
`;

exports[`Input.Idcard renders correctly 1`] = `
<Input
  type="idcard"
>
  <InputNumber
    disabled={false}
    prefixCls="za-input"
    type="idcard"
  >
    <div
      className="za-input za-input-number"
      onClick={[Function]}
    >
      <div
        className="za-input-placeholder"
      />
      <div
        className="za-input-content"
      />
      <input
        disabled={false}
        type="hidden"
        value=""
      />
      <KeyboardPicker
        onKeyClick={[Function]}
        prefixCls="za-keyboard"
        type="idcard"
        visible={false}
      >
        <div
          className="za-keyboard-picker"
        >
          <Popup
            animationDuration={200}
            autoClose={false}
            direction="bottom"
            mask={false}
            maskType="normal"
            prefixCls="za-popup"
            stayTime={3000}
            visible={false}
          >
            <div>
              <Portal
                getContainer={[Function]}
              >
                <div
                  className="za-popup za-popup-hidden"
                >
                  <div
                    className="za-popup-wrapper za-popup-wrapper-bottom"
                    style={
                      Object {
                        "WebkitTransitionDuration": "200ms",
                        "transitionDuration": "200ms",
                      }
                    }
                  >
                    <Keyboard
                      onKeyClick={[Function]}
                      prefixCls="za-keyboard"
                      type="idcard"
                      visible={false}
                    >
                      <div
                        className="za-keyboard"
                        onClick={[Function]}
                      >
                        <div
                          className="za-keyboard-keys"
                        >
                          <div
                            className="za-keyboard-item"
                            key="0"
                            onClick={[Function]}
                          >
                            1
                          </div>
                          <div
                            className="za-keyboard-item"
                            key="1"
                            onClick={[Function]}
                          >
                            2
                          </div>
                          <div
                            className="za-keyboard-item"
                            key="2"
                            onClick={[Function]}
                          >
                            3
                          </div>
                          <div
                            className="za-keyboard-item"
                            key="3"
                            onClick={[Function]}
                          >
                            4
                          </div>
                          <div
                            className="za-keyboard-item"
                            key="4"
                            onClick={[Function]}
                          >
                            5
                          </div>
                          <div
                            className="za-keyboard-item"
                            key="5"
                            onClick={[Function]}
                          >
                            6
                          </div>
                          <div
                            className="za-keyboard-item"
                            key="6"
                            onClick={[Function]}
                          >
                            7
                          </div>
                          <div
                            className="za-keyboard-item"
                            key="7"
                            onClick={[Function]}
                          >
                            8
                          </div>
                          <div
                            className="za-keyboard-item"
                            key="8"
                            onClick={[Function]}
                          >
                            9
                          </div>
                          <div
                            className="za-keyboard-item"
                            key="9"
                            onClick={[Function]}
                          >
                            x
                          </div>
                          <div
                            className="za-keyboard-item"
                            key="10"
                            onClick={[Function]}
                          >
                            0
                          </div>
                          <div
                            className="za-keyboard-item"
                            key="11"
                            onClick={[Function]}
                          >
                            <Icon
                              prefixCls="za-icon"
                              type="keyboard"
                            >
                              <i
                                className="za-icon za-icon-keyboard"
                              />
                            </Icon>
                          </div>
                        </div>
                        <div
                          className="za-keyboard-handle"
                        >
                          <div
                            className="za-keyboard-item"
                            onTouchEnd={[Function]}
                            onTouchStart={[Function]}
                          >
                            <Icon
                              prefixCls="za-icon"
                              type="deletekey"
                            >
                              <i
                                className="za-icon za-icon-deletekey"
                              />
                            </Icon>
                          </div>
                          <div
                            className="za-keyboard-item za-keyboard-item-ok"
                            onClick={[Function]}
                          >
                            确定
                          </div>
                        </div>
                      </div>
                    </Keyboard>
                  </div>
                </div>
              </Portal>
            </div>
          </Popup>
        </div>
      </KeyboardPicker>
    </div>
  </InputNumber>
</Input>
`;

exports[`Input.Number renders correctly 1`] = `
<Input
  type="number"
>
  <InputNumber
    disabled={false}
    prefixCls="za-input"
    type="number"
  >
    <div
      className="za-input za-input-number"
      onClick={[Function]}
    >
      <div
        className="za-input-placeholder"
      />
      <div
        className="za-input-content"
      />
      <input
        disabled={false}
        type="hidden"
        value=""
      />
      <KeyboardPicker
        onKeyClick={[Function]}
        prefixCls="za-keyboard"
        type="number"
        visible={false}
      >
        <div
          className="za-keyboard-picker"
        >
          <Popup
            animationDuration={200}
            autoClose={false}
            direction="bottom"
            mask={false}
            maskType="normal"
            prefixCls="za-popup"
            stayTime={3000}
            visible={false}
          >
            <div>
              <Portal
                getContainer={[Function]}
              >
                <div
                  className="za-popup za-popup-hidden"
                >
                  <div
                    className="za-popup-wrapper za-popup-wrapper-bottom"
                    style={
                      Object {
                        "WebkitTransitionDuration": "200ms",
                        "transitionDuration": "200ms",
                      }
                    }
                  >
                    <Keyboard
                      onKeyClick={[Function]}
                      prefixCls="za-keyboard"
                      type="number"
                      visible={false}
                    >
                      <div
                        className="za-keyboard"
                        onClick={[Function]}
                      >
                        <div
                          className="za-keyboard-keys"
                        >
                          <div
                            className="za-keyboard-item"
                            key="0"
                            onClick={[Function]}
                          >
                            1
                          </div>
                          <div
                            className="za-keyboard-item"
                            key="1"
                            onClick={[Function]}
                          >
                            2
                          </div>
                          <div
                            className="za-keyboard-item"
                            key="2"
                            onClick={[Function]}
                          >
                            3
                          </div>
                          <div
                            className="za-keyboard-item"
                            key="3"
                            onClick={[Function]}
                          >
                            4
                          </div>
                          <div
                            className="za-keyboard-item"
                            key="4"
                            onClick={[Function]}
                          >
                            5
                          </div>
                          <div
                            className="za-keyboard-item"
                            key="5"
                            onClick={[Function]}
                          >
                            6
                          </div>
                          <div
                            className="za-keyboard-item"
                            key="6"
                            onClick={[Function]}
                          >
                            7
                          </div>
                          <div
                            className="za-keyboard-item"
                            key="7"
                            onClick={[Function]}
                          >
                            8
                          </div>
                          <div
                            className="za-keyboard-item"
                            key="8"
                            onClick={[Function]}
                          >
                            9
                          </div>
                          <div
                            className="za-keyboard-item za-keyboard-item-disabled"
                            key="9"
                            onClick={[Function]}
                          />
                          <div
                            className="za-keyboard-item"
                            key="10"
                            onClick={[Function]}
                          >
                            0
                          </div>
                          <div
                            className="za-keyboard-item"
                            key="11"
                            onClick={[Function]}
                          >
                            <Icon
                              prefixCls="za-icon"
                              type="keyboard"
                            >
                              <i
                                className="za-icon za-icon-keyboard"
                              />
                            </Icon>
                          </div>
                        </div>
                        <div
                          className="za-keyboard-handle"
                        >
                          <div
                            className="za-keyboard-item"
                            onTouchEnd={[Function]}
                            onTouchStart={[Function]}
                          >
                            <Icon
                              prefixCls="za-icon"
                              type="deletekey"
                            >
                              <i
                                className="za-icon za-icon-deletekey"
                              />
                            </Icon>
                          </div>
                          <div
                            className="za-keyboard-item za-keyboard-item-ok"
                            onClick={[Function]}
                          >
                            确定
                          </div>
                        </div>
                      </div>
                    </Keyboard>
                  </div>
                </div>
              </Portal>
            </div>
          </Popup>
        </div>
      </KeyboardPicker>
    </div>
  </InputNumber>
</Input>
`;

exports[`Input.Price renders correctly 1`] = `
<Input
  type="price"
>
  <InputNumber
    disabled={false}
    prefixCls="za-input"
    type="price"
  >
    <div
      className="za-input za-input-number"
      onClick={[Function]}
    >
      <div
        className="za-input-placeholder"
      />
      <div
        className="za-input-content"
      />
      <input
        disabled={false}
        type="hidden"
        value=""
      />
      <KeyboardPicker
        onKeyClick={[Function]}
        prefixCls="za-keyboard"
        type="price"
        visible={false}
      >
        <div
          className="za-keyboard-picker"
        >
          <Popup
            animationDuration={200}
            autoClose={false}
            direction="bottom"
            mask={false}
            maskType="normal"
            prefixCls="za-popup"
            stayTime={3000}
            visible={false}
          >
            <div>
              <Portal
                getContainer={[Function]}
              >
                <div
                  className="za-popup za-popup-hidden"
                >
                  <div
                    className="za-popup-wrapper za-popup-wrapper-bottom"
                    style={
                      Object {
                        "WebkitTransitionDuration": "200ms",
                        "transitionDuration": "200ms",
                      }
                    }
                  >
                    <Keyboard
                      onKeyClick={[Function]}
                      prefixCls="za-keyboard"
                      type="price"
                      visible={false}
                    >
                      <div
                        className="za-keyboard"
                        onClick={[Function]}
                      >
                        <div
                          className="za-keyboard-keys"
                        >
                          <div
                            className="za-keyboard-item"
                            key="0"
                            onClick={[Function]}
                          >
                            1
                          </div>
                          <div
                            className="za-keyboard-item"
                            key="1"
                            onClick={[Function]}
                          >
                            2
                          </div>
                          <div
                            className="za-keyboard-item"
                            key="2"
                            onClick={[Function]}
                          >
                            3
                          </div>
                          <div
                            className="za-keyboard-item"
                            key="3"
                            onClick={[Function]}
                          >
                            4
                          </div>
                          <div
                            className="za-keyboard-item"
                            key="4"
                            onClick={[Function]}
                          >
                            5
                          </div>
                          <div
                            className="za-keyboard-item"
                            key="5"
                            onClick={[Function]}
                          >
                            6
                          </div>
                          <div
                            className="za-keyboard-item"
                            key="6"
                            onClick={[Function]}
                          >
                            7
                          </div>
                          <div
                            className="za-keyboard-item"
                            key="7"
                            onClick={[Function]}
                          >
                            8
                          </div>
                          <div
                            className="za-keyboard-item"
                            key="8"
                            onClick={[Function]}
                          >
                            9
                          </div>
                          <div
                            className="za-keyboard-item"
                            key="9"
                            onClick={[Function]}
                          >
                            .
                          </div>
                          <div
                            className="za-keyboard-item"
                            key="10"
                            onClick={[Function]}
                          >
                            0
                          </div>
                          <div
                            className="za-keyboard-item"
                            key="11"
                            onClick={[Function]}
                          >
                            <Icon
                              prefixCls="za-icon"
                              type="keyboard"
                            >
                              <i
                                className="za-icon za-icon-keyboard"
                              />
                            </Icon>
                          </div>
                        </div>
                        <div
                          className="za-keyboard-handle"
                        >
                          <div
                            className="za-keyboard-item"
                            onTouchEnd={[Function]}
                            onTouchStart={[Function]}
                          >
                            <Icon
                              prefixCls="za-icon"
                              type="deletekey"
                            >
                              <i
                                className="za-icon za-icon-deletekey"
                              />
                            </Icon>
                          </div>
                          <div
                            className="za-keyboard-item za-keyboard-item-ok"
                            onClick={[Function]}
                          >
                            确定
                          </div>
                        </div>
                      </div>
                    </Keyboard>
                  </div>
                </div>
              </Portal>
            </div>
          </Popup>
        </div>
      </KeyboardPicker>
    </div>
  </InputNumber>
</Input>
`;

exports[`Input.Textarea autoHeight 1`] = `
<Input
  autoHeight={true}
  onChange={[Function]}
  rows={4}
  type="textarea"
  value="foo"
>
  <InputTextarea
    autoHeight={true}
    disabled={false}
    focused={false}
    onChange={[Function]}
    prefixCls="za-input"
    rows={4}
    showLength={false}
    value="foo"
  >
    <div
      className="za-input za-input-textarea"
    >
      <textarea
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        rows={4}
        value="foo"
      />
    </div>
  </InputTextarea>
</Input>
`;

exports[`Input.Textarea renders correctly 1`] = `
<div
  class="za-input za-input-textarea"
>
  <textarea
    rows="4"
  />
</div>
`;
